api/src/services/fields.ts

```diff
@@ -488,6 +478,16 @@ export class FieldsService {
 			}
 
 			await trx('directus_fields').delete().where({ collection, field });
+
+			if (
+				this.schema.collections[collection] &&
+				field in this.schema.collections[collection].fields &&
+				this.schema.collections[collection].fields[field].alias === false
+			) {
+				await trx.schema.table(collection, (table) => {
+					table.dropColumn(field);
+				});
+			}
```

Comment:
This shouldn't work on CockroachDB. You can't do any DDL operations after a regular SQL query 🤔

Reply:
Ah bummer. Somehow it doesn't show any error for me when I tested it on the CockroachDB docker image, but the scenario I tested was probably too rudimentary. Honestly not sure what would the fix be now 🤔

Reply:
Maybe adding a manual `dropForeign` before the `dropColumn` to force delete the constraint no matter what before dropping the column? 🤔

Reply:
The problem here was that the database constraint was being deleted by dropColumn (but not in every DB), which caused the following `dropForeign` call to fail (which happens in the `relationsService.deleteOne()` call. My fix is to explicitly check if the constraint still exists before attempting to delete it in the relations service, as trying to delete a non-existing key causes a database error 👍🏻